### ISSUE TITLE
Wirecard_ElasticEngine overriding Sofort_Payment virtualType

### ISSUE THREAD

**Issue by gego05** [wirecard/magento2-ee#180]
### Versions
Magento version: 2.1.6
PHP version: 7.2.12

### Issue
We have currently installed your payment modul, and after that, our sofort payment selection in checkout is not visible any more. So after a little debugging, we noticed that Wirecard_ElasticEngine creates a virtualType called "SofortFacade" in di.xml.

The problem is, that Sofort_Payment (https://www.sofort.com/integrationCenter-ger-DE/integration/shopsysteme/Magento-Community/#14098) creates the same virtualType called "SofortFacade".
So, your module is overriding the virtualType and if we activate your module, we can't use the sofort payment from klarna anymore. If we deactivate it, everything is working fine.

Is there a reason for that, or can we just rename the virtualType?

### Steps to reproduce
1) Install Sofort_Payment (https://www.sofort.com/integrationCenter-ger-DE/integration/shopsysteme/Magento-Community/#14098)
2) Activate Sofort_Payment so you can see it in checkout
3) Install Wirecard_ElasticEngine and activate it
4) Go to checkout, sofort payment is not visible anymore

**Comment by rinnhofer** [comment]
Hello @gego05
sorry for the delayed response to your issue.
Unfortunately Sofort_Payment is not pre-installed within magento2 and considered as third-party-plugin therefor we did not include it in our standard tests during development.

Thank you for providing the integration document for Sofort_Payment and the solution idea. After gathering a test account for Sofort_Payment and reproduction according to your steps we can confirm that Sofort_Payment and our Implementation are not compatible with each other. We tried to resolve the incompatibility with the suggested solution but sadly it did not work out.

We will try to figure something out in the future, but finding a solution which works for both sides will take some time an planning on our site.

For the moment we can not provide a date for a possible solution but there is always the option to use Sofort. within our implementation until then.

Kind regards,
Wirecard Plugin Team
